src/main/scala/io/epiphanous/flinkrunner/util/AvroUtils.scala

```diff
@@ -104,6 +104,7 @@ object AvroUtils extends LazyLogging {
           headers
         )
       )
+  }
 
   implicit class RichGenericRecord(genericRecord: GenericRecord) {
```

Comment:
I think you can replace the whole toSpecific method with this:

` def toSpecific[A <: GenericRecord](instance: A): A = 
      SpecificData.get().deepCopy(instance.getSchema, genericRecord).asInstanceOf[A]`

Reply:
I tested it briefly and it seems to work

Reply:
Seems to work meaning, the way it's working now.  I do have an issue though with logical type conversions.  Still trying to figure it out.

Reply:
That's brilliant! Good find.

Reply:
Given that this can fail, we should wrap it in a `Try`, no?

Reply:
Yes!